test/appium/views/chat_view.py

```diff
@@ -186,6 +187,22 @@ def wait_for_messages_by_user(self, username: str, expected_messages: list, erro
             errors.append('Not received messages from user %s: "%s"' % (username, ', '.join(
                 [i for i in list(set(expected_messages) - set(received_messages))])))
 
+    def wait_for_messages(self, username: str, expected_messages: list, errors: list, wait_time: int = 30):
+        expected_messages = expected_messages if type(expected_messages) == list else [expected_messages]
+        repeat = 0
+        received_messages = list()
+        while repeat <= wait_time:
+            for message in expected_messages:
+                if self.element_by_text(message, 'text').is_element_present(1):
+                    received_messages.append(message)
+            if not set(expected_messages) - set(received_messages):
+                break
+            time.sleep(3)
+            repeat += 3
+        if set(expected_messages) - set(received_messages):
+            errors.append('Not received messages from user %s: "%s"' % (username, ', '.join(
```

Comment:
Maybe we should raise an exception here instead of appending it to some external variable.

Reply:
As it's used in a quite long and complex test we are doing this to gather all errors together and raise exception after all verifications